lib/Attribute.js

```diff
@@ -196,42 +197,54 @@ Attribute.prototype.types = {
   }
 };
 
+// This function takes in a `value` as a parameter and sets the type of this attribute based on the type of that value.
+// No type defined - assume this is not a type definition and we must grab type directly from value.
+// Examples of `value` parameters that might be passed in:
+// { type: [Function: String], required: true, default: 'Mittens' }
+// function String() { [native code] }
+// { type: [Function: Number], validate: [Function: validate], default: 888 }
+// 1
+// false
+// "puddles"
+// [ 'cute', 'fuzzy' ]
+// { name: 'mittens', age: 1 }
 Attribute.prototype.setTypeFromRawValue = function(value) {
-  //no type defined - assume this is not a type definition and we must grab type directly from value
   let type;
-  let typeVal = value;
-  if (value.type){
-    typeVal = value.type;
-  }
-
-  if (Array.isArray(typeVal) || typeVal === 'list'){
-    type = 'List';
-  } else if ( (Array.isArray(typeVal) && typeVal.length === 1) || typeof typeVal === 'function') {
-    this.isSet = Array.isArray(typeVal);
-    let regexFuncName = /^Function ([^(]+)\(/i;
-    let found = typeVal.toString().match(regexFuncName);
-    type = found[1];
-    if (type === 'Object') {
-      type = 'Map';
-    }
-  } else if (typeof typeVal === 'object' || typeVal === 'map'){
-    type = 'Map';
+  const typeVal = value.type || value;
+
+  if (typeof typeVal === 'function') {
```

Comment:
Could we use any of the lodash helpers here?

[lang helpers](https://lodash.com/docs/4.17.11#isArray)

Reply:
@hweeks `lodash` is not a dependency of this project. It was removed in a PR a decent amount of time ago I think.

Reply:
i was thinking it may be worth using for these types of checks. it's more robust than a regex, and generally I'd have more faith in it. On top of that, this lib is almost always running on node, so the bundle size isn't as vital.

Reply:
@hweeks I don't think this will work:

```
> _.isNumber(Number)
false
> _.isNumber(5)
true
```

Reply:
ahh, i misread what this was doing. I got it now. I was using lodash elsewhere in a plugin i was writing. Lemme read through this more thoroughly